controller/controller.go

```diff
@@ -272,6 +349,14 @@ func deleteSchedulerHelper(logger logrus.FieldLogger, mr *models.MixedMetricsRep
 		}
 	}
 
+	err = mr.WithSegment(models.SegmentDelete, func() error {
```

Comment:
Why did you move the scheduler removal from db to the first operation? It was the last beacause if something failed, the watcher would recreate the other parts (namespace, redis info, etc) and if the user still wanted to delete the scheduler he could call the endpoint again.

If you first delete it from the db and then attempt to do the remaining ops when there's a failure we'll have junk info (maybe a namespace, junk info in redis, etc) that we won't be able to detect easily.

Please move it back to the last op and add a comment just above it explaining why this is needed.